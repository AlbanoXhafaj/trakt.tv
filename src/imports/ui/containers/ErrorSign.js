import React from "react";

//a red cycle with a big X in middle
const ErrorSign = () =>(
  <div className ='error-circle'>
    <div>X</div>
  </div>
);

export default ErrorSign;
