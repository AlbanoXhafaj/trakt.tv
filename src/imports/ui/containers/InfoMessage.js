import React from "react";
import PropTypes from 'prop-types';
// a centered div with a text displaying an info
const InfoMessage = ({ children }) => (
  <div className="infoMessage">
    <h3>
      { children }
    </h3>
  </div>
)

InfoMessage.propTypes = {
  children: PropTypes.string.isRequired
};

export default InfoMessage;
