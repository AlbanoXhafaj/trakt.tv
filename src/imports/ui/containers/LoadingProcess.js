import React from "react";

//Loading Cycle during the moment that data is fetching
const LoadingProcess = () => (
  <div className="loader"></div>
);

export default LoadingProcess;
