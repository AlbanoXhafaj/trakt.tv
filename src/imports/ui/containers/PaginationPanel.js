import React from "react";
import PropTypes from 'prop-types';
import { connect } from "react-redux";

import * as actionCreators from "actions/actionCreators";
import * as selectors from "store/stateSelectors";
// a hyperlink used as button to load more indexes in descending order.
export const LeftDoubleArrow = ({ displayedIndexes , startIndex , loadLeftPaginationIndexes }) => (
  <a
    href=""
    onClick = { (e) => { e.preventDefault(); loadLeftPaginationIndexes( startIndex , displayedIndexes ); }}
  >
    {"<<"}
  </a>
)

// a hyperlink used as button to load more indexes in ascending order.
export const RightDoubleArrow = ({ displayedIndexes , startIndex , pagesNumber , loadRightPaginationIndexes }) => (
  <a
    href=""
    onClick = { (e) => { e.preventDefault(); loadRightPaginationIndexes( startIndex , pagesNumber , displayedIndexes); }}
  >
    {">>"}
  </a>
)

//a hyperlink representin displayed indexes that noone is the active one, on click passes index as param
export const Index = ({ onChange ,index }) => (
  <a
    href=""
    onClick = { (e) => { e.preventDefault(); onChange(index); } }
  >
    { index }
  </a>
)

//the only active link represent the displayed page index.
export const ActiveIndex = ({ index }) => (
  <a
    href=""
    className="active"
    onClick = { (e) => { e.preventDefault(); }}
  >
    { index }
  </a>
)

// a function that stores all the generated indexes in an array and returns them wrapped in a div.
export const pageIndexes = ({ displayedIndexes , pageIndex , pagesNumber , startIndex , switchPage } ) => {
  let indexes=[];
  for( let i=0; i< displayedIndexes; i++ ){
    let newIndex = startIndex + i;
    if( newIndex <= pagesNumber) {

      if( newIndex === pageIndex )
        indexes.push( <ActiveIndex key={i}  index={ newIndex } /> );
      else
        indexes.push( <Index key={i} onChange= { switchPage } index={ newIndex } /> );
    }
  }

  return <div>{indexes}</div>;
}

export const _PaginationPanel = (props) => (
  <div className="pagination">
    <LeftDoubleArrow {...props} />
    { pageIndexes(props) }
    <RightDoubleArrow  {...props}/>
  </div>
)

const mapStateToProps = (state) =>({
  pageIndex: selectors.getPageIndex(state),
  pagesNumber: selectors.getPagesNumber(state),
  startIndex: selectors.getStartIndex(state),
  displayedIndexes: selectors.getDisplayedIndexes(state)
});

const mapDispatchToProps = {
  loadLeftPaginationIndexes: actionCreators.loadLeftPaginationIndexes,
  loadRightPaginationIndexes: actionCreators.loadRightPaginationIndexes,
  switchPage: actionCreators.switchPage
};

const PaginationPanel = connect(mapStateToProps,mapDispatchToProps)(_PaginationPanel);

PaginationPanel.propTypes = {
  pageIndex: PropTypes.number.isRequired,
  pagesNumber: PropTypes.number.isRequired,
  startIndex: PropTypes.number.isRequired,
  displayedIndexes: PropTypes.number.isRequired,
  loadLeftPaginationIndexes: PropTypes.func.isRequired,
  loadRightPaginationIndexes: PropTypes.func.isRequired,
  switchPage: PropTypes.func.isRequired
};

export default PaginationPanel;
