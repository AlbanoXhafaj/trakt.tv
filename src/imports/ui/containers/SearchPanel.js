import React from "react";
import PropTypes from 'prop-types';

// OnChange pass the text of input as parameter
// Input box value equals to the search filters text, needed for wipe out by other actions
const SearchPanel = ( { onChange, text }) => (
  <div className="inline">
    <span>Search:</span>
    <input
      type="text"
      placeholder="keyword..."
      value = { text }
      onChange = { (e) => onChange(e.target.value) }
    />
  </div>
);

SearchPanel.propTypes = {
  onChange: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired
};

export default SearchPanel;
