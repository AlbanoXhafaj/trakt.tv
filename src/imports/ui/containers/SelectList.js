import React from "react";
import PropTypes from 'prop-types';

//DropdownList generate by an array of strings
//title - describes the dropwdownlist user
//options - array of strings, elements of the list
//onChange - function that will accept as a parametre the index of the element
//fistElement - the first element of the list like "select options below"

const SelectList = ({ title, options , onChange , firstElement = null }) => (
  <div className="inline">
    <span> { title }: </span>
    <select onChange = { (e) => onChange( parseInt(e.target.value, 10) ) } >
      {
        firstElement
          ? <option value="-1"> { firstElement } </option>
          :null
      }
      {
        //create <option> tags with index as value
        options.map( (option,index) => (
          <option
            key= { index }
            value = { index }
          >
            { option }
          </option>
        ))
      }
    </select>
  </div>
);

SelectList.propTypes = {
  title: PropTypes.string.isRequired,
  options: PropTypes.array.isRequired,
  onChange: PropTypes.func.isRequired,
  fistElement: PropTypes.string
};

export default SelectList;
