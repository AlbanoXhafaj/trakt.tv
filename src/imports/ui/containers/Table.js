import React from "react";
import PropTypes from 'prop-types';
import { connect } from "react-redux";


//It splits the shows array in a matrix[  rows ,columnCount]
const splitInRows = (shows, columnCount) => {
  let rows = [];
  let showsCopy = [ ...shows ];

  while( showsCopy.length > 0 ) rows.push( showsCopy.splice(0,columnCount) );

  return rows;
};

const Row = (shows,index) => (
  <tr key= { index } >
    {
      shows.map( (show,index) => Cell(show,index)) //for every show in row we create a cell
    }
  </tr>
);

const Cell = (show,index) =>(
  <td key={index}>
    <div className="cellTitle">
      <span><b>{ show.title }</b></span>
    </div>
    <div className="cellBody">
      <img src = { show.poster } alt={ show.title +" 's poster"} />
    </div>
    <div className="cellFooter">
      <span>Trakt Id: { show.ids.trakt }</span><br />
      <span>Rating: { Math.round( show.rating * 10) / 10 }</span><br />
      <span>Year: { show.year }</span><br/>
      <span>Episodes: { show.aired_episodes } </span>
    </div>
  </td>
);

export const _Table = ({ shows, columnCount }) => {

  const showsPerRow = splitInRows( shows , columnCount);

  return(
    <table className="center">
      <tbody>
      {
          showsPerRow.map( (row,index) => Row(row,index) ) //for every row we create a Row container
      }
      </tbody>
    </table>
  )
}

const mapStateToProps = (state) => ({
  shows: state.fetch.data.shows,
  columnCount: state.filters.gridSizeFilter.columns
});

const Table = connect( mapStateToProps , null )(_Table);

Table.propTypes = {
  shows: PropTypes.arrayOf(PropTypes.object).isRequired,
  columnCount: PropTypes.number.isRequired
};

export default Table;
