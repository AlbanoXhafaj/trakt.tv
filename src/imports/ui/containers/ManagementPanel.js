import React from "react";
import PropTypes from 'prop-types';

import SelectList from "containers/SelectList";
import SearchPanel from "containers/SearchPanel";

import * as actionCreators from "actions/actionCreators";
import { filters } from "reducers/FiltersReducer";
import * as selectors from "store/stateSelectors";
import { connect } from "react-redux";

//Creates 2 SelectLists, 1 SearchPanel and if needed 1 extra Select List for the period filters
export const _ManagementPanel = ({ sortFilter, searchText , setSortFilter , setPeriodFilter, setGridSizeFilter , setSearchFilter }) => (
  <fieldset>
    <legend>Management Panel</legend>
    <SelectList
      title="Sort"
      options = { filters.sortFilters.getValues()  }
      onChange = { setSortFilter }
    />
    {
      sortFilter.needPeriodParameter
        ? <SelectList
            title="Period"
            options = { filters.periodFilters.getValues() }
            onChange = { setPeriodFilter }
          />
        : null
    }
    <SearchPanel onChange = { setSearchFilter } text = { searchText }/>
    <SelectList
      title="Grid Size"
      options = { filters.gridSizeFilters.getValues() }
      onChange = { setGridSizeFilter }
    />
  </fieldset>
);

export const mapStateToProps = (state) =>({
  sortFilter: selectors.getSortFilter(state),
  searchText: selectors.getSearchText(state)
});

export const mapDispatchToProps = {
  setSortFilter: actionCreators.setSortFilter,
  setPeriodFilter: actionCreators.setPeriodFilter,
  setGridSizeFilter: actionCreators.setGridSizeFilter,
  setSearchFilter: actionCreators.setSearchFilter
};

const ManagementPanel = connect(mapStateToProps,mapDispatchToProps)(_ManagementPanel);

ManagementPanel.propTypes = {
  sortFilter: PropTypes.exact({
    url: PropTypes.string,
    value: PropTypes.string,
    needPeriodParameter: PropTypes.bool
  }).isRequired,
  searchText: PropTypes.string.isRequired,
  searchFilter: PropTypes.func.isRequired,
  setPeriodFilter: PropTypes.func.isRequired,
  setGridSizeFilter: PropTypes.func.isRequired,
  setSearchFilter: PropTypes.func.isRequired
};

export default ManagementPanel;
