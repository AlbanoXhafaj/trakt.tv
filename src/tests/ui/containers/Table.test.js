import "../../setups/setup_enzyme.js";
import React from "react";

import { shallow , mount } from "enzyme";
import toJSON from 'enzyme-to-json';

import { _Table } from "containers/Table";

describe("<Table />", () => {
  const initialProps = {
    shows: [
      {
        ids: { trakt: "1"},
        title: "Show 1",
        poster: "Show1.jpg",
        rating: 7.56732,
        year: 1990,
        aired_episodes: 5
      },
      {
        ids: { trakt: "2"},
        title: "Show 2",
        poster: "Show2.jpg",
        rating: 8.3431,
        year: 2011,
        aired_episodes: 17
      },
      {
        ids: { trakt: "3"},
        title: "Show 3",
        poster: "Show3.jpg",
        rating: 6.87932,
        year: 2003,
        aired_episodes: 32
      }
    ],
    columnCount: 3
  }

  const wrapper = shallow(<_Table {...initialProps} />);

  it("should match the snapshot" , () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
