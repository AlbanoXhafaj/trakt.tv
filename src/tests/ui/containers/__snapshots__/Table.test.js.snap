// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should match the snapshot 1`] = `
<table
  className="center"
>
  <tbody>
    <tr
      key="0"
    >
      <td
        key="0"
      >
        <div
          className="cellTitle"
        >
          <span>
            <b>
              Show 1
            </b>
          </span>
        </div>
        <div
          className="cellBody"
        >
          <img
            alt="Show 1 's poster"
            src="Show1.jpg"
          />
        </div>
        <div
          className="cellFooter"
        >
          <span>
            Trakt Id: 
            1
          </span>
          <br />
          <span>
            Rating: 
            7.6
          </span>
          <br />
          <span>
            Year: 
            1990
          </span>
          <br />
          <span>
            Episodes: 
            5
             
          </span>
        </div>
      </td>
      <td
        key="1"
      >
        <div
          className="cellTitle"
        >
          <span>
            <b>
              Show 2
            </b>
          </span>
        </div>
        <div
          className="cellBody"
        >
          <img
            alt="Show 2 's poster"
            src="Show2.jpg"
          />
        </div>
        <div
          className="cellFooter"
        >
          <span>
            Trakt Id: 
            2
          </span>
          <br />
          <span>
            Rating: 
            8.3
          </span>
          <br />
          <span>
            Year: 
            2011
          </span>
          <br />
          <span>
            Episodes: 
            17
             
          </span>
        </div>
      </td>
      <td
        key="2"
      >
        <div
          className="cellTitle"
        >
          <span>
            <b>
              Show 3
            </b>
          </span>
        </div>
        <div
          className="cellBody"
        >
          <img
            alt="Show 3 's poster"
            src="Show3.jpg"
          />
        </div>
        <div
          className="cellFooter"
        >
          <span>
            Trakt Id: 
            3
          </span>
          <br />
          <span>
            Rating: 
            6.9
          </span>
          <br />
          <span>
            Year: 
            2003
          </span>
          <br />
          <span>
            Episodes: 
            32
             
          </span>
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
